include/ghex/transport_layer/context.hpp

```diff
@@ -191,19 +78,28 @@ namespace gridtools {
                 // thread-safe
                 thread_token get_token() noexcept
                 {
-                    return m_parallel_context.m_thread_primitives.get_token();
+                    return m_thread_primitives.get_token();
                 }
-                
+
                 // thread-safe
                 void barrier(thread_token& t) /*const*/
                 {
-                    m_parallel_context.barrier(t);
+                    m_thread_primitives.barrier(t);
```

Comment:
@boeschf The barrier has been moved to the transport context, as we discussed. I still need to implement a trivial barrier for UCX transport.